proposals/manual-bundling.md

```diff
@@ -0,0 +1,150 @@
+
+   Now this is where things get weird.
+
+   A directory `/platforms/android/platform_www` is created.
+
+   Inside this folder is a file `cordova.js`, which contains a UMD bundle with the following modules:
+
+   - `'cordova'` → https://github.com/apache/cordova-js/blob/master/src/cordova.js
+   - `'cordova/exec'` → https://github.com/apache/cordova-android/blob/master/cordova-js-src/exec.js
+   - `'cordova/platform'` → https://github.com/apache/cordova-android/blob/master/cordova-js-src/platform.js
+   - `'cordova/android/*'` → https://github.com/apache/cordova-android/tree/master/cordova-js-src/android
+   - `'cordova/plugin/android/*'` → https://github.com/apache/cordova-android/tree/master/cordova-js-src/plugin/android
+   - `'cordova/*'` → https://github.com/apache/cordova-js/tree/master/src/common (except for exec and platform, not sure if they're the `_b` postfixed ones, but it doesn't really matter)
+
+   In addition there is a `cordova_plugins.js`, which maps the plugins to a file location.
+
+   Also there is a `plugins/cordova-plugin-camera/www/*`, which contains the files from https://github.com/apache/cordova-plugin-camera/tree/master/www, but with a `.` as a path separator, whereas in any other situation this would be a `/`.
+
+   However, if I take a look at what the `cordova` package on npm is, it's the CLI, which is totally unrelated to the UMD bundle.
+   
+   In addition to `/platforms`, there's also a `/plugins` directory. This contains the following:
+   
+   - Copies of the plugins that already exist in `node_modules`, but with the dotfiles missing.
+   - `android.json` — This appears to duplicate some info from `package.json` and `config.xml`, but in another format.
+   - `fetch.json` — This appears to duplicate some info from `package.json`, but in yet another format.
+
+
+## Proposal
+
+As a modern JavaScript developer, I would like to use *<this year's popular tool>* to bundle my dependencies and code.
+
+- Get rid of `cordova.js`.
+- Get rid of `www/`.
+- Get rid of `plugins/`.
+
+### 1. Load plugins from the `node_modules` directory, not `plugins`
+
+Cordova already downloads packages to `node_modules` using `cordova-fetch`. There's no need to maintain a copy of them in the `plugins` directory.
+
+This also allows plugin creators to use npm dependencies for both hooks and browser code the way this would be done for other packages.
+
+### 2. Let a developer require Cordova using its own build tools.
+
+Based on [cordova-plugin-camera] as an example:
+
+As a developer, I'd like to turn this:
+
+```js
+navigator.camera.getPicture(cameraSuccess, cameraError, cameraOptions)
+```
+
+Into this:
+
+```js
+import CordovaCamera from 'cordova-plugin-camera';
+
+
+CordovaCamera.getPicture(cameraSuccess, cameraError, cameraOptions);
+```
+
+This can be done by adding the following to `package.json`:
+
+```json
+  "main": "www/Camera.js"
+```
+
+### 3. Use [cordova-js] as a proxy to the platform
```

Comment:
cordova-js definitely needs to be rethought out. This is an interesting idea I'd like to hear more discussion about.